docs/scripts/link-checker-puppeteer.ts

```diff
@@ -0,0 +1,72 @@
+import { PromisePool } from '@supercharge/promise-pool';
+import { sitePaths } from '../src/data/sitePaths';
+import { checkLink, crawlAllLinks } from './util';
+import type { LinkInfo } from './util';
+
+/**
+ * Divide the sitePaths array so that we can easily run a smaller portion if needed.
+ * e.g. to run the first 10 links. run `yarn docs node --require esbuild-register ./scripts/link-checker-puppeteer.ts 0 10`
+ */
+const start = process.argv[2] ?? 0;
+const end = process.argv[3];
+const testPaths = end ? sitePaths.slice(+start, +end) : sitePaths.slice(+start);
+
+try {
+  runLinkChecker();
+} catch (err) {
+  process.exit(1);
+}
+
+async function runLinkChecker() {
+  const allPagesPaths = await crawlAllLinks(testPaths);
+
+  const { results } = await PromisePool.withConcurrency(10)
+    .for(allPagesPaths)
+    .process(async (pagePaths, pageIdx, pool) => {
+      const { pageUrl, links } = pagePaths;
+      const { results } = await PromisePool.withConcurrency(10)
```

Comment:
nit: would be nice to have the number of concurrent processes as a const, so we can modify it in one place if needed in the future. Also a note on why we landed on 10 and  if there are any limitations increasing it. As we will have more links we might need to revisit at some point.

Reply:
Done 13070f2bc58675243a33030366b92eb7a03fa34b